DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/UIImage+SVG.swift

```diff
@@ -0,0 +1,178 @@
+    }
+}
+
+private struct SVGShapeDocument {
+    private final class ParserDelegate: NSObject, XMLParserDelegate {
+        var svgAttributes: [String: String] = [:]
+        var pathAttributes: [String: String] = [:]
+
+        private var didParseSVG = false
+        private var didParsePath = false
+
+        func parser(
+            _ parser: XMLParser,
+            didStartElement elementName: String,
+            namespaceURI: String?,
+            qualifiedName qName: String?,
+            attributes attributeDict: [String: String] = [:]
+        ) {
+            switch elementName.lowercased() {
+            case "svg":
+                guard !didParseSVG else {
+                    return
+                }
+                svgAttributes = attributeDict
+                didParseSVG = true
+            case "path":
+                guard !didParsePath else {
+                    return
+                }
+                pathAttributes = attributeDict
+                didParsePath = true
+            default:
+                break
+            }
+        }
+    }
+
+    let size: CGSize
+    let path: CGPath
+    let fillColor: UIColor
+    let isEOFilled: Bool
+
+    init?(svgData: Data) {
+        let delegate = ParserDelegate()
+
+        let parser = XMLParser(data: svgData)
+        parser.delegate = delegate
+        parser.shouldResolveExternalEntities = false
+
+        guard parser.parse() else {
+            return nil
+        }
+
+        let numberFormatter = NumberFormatter()
+        numberFormatter.locale = Locale(identifier: "en_US_POSIX")
+        numberFormatter.numberStyle = .decimal
+        numberFormatter.allowsFloats = true
+
+        guard
+            let width = delegate.svgAttributes["width"].flatMap(numberFormatter.number(from:))?.doubleValue,
+            let height = delegate.svgAttributes["height"].flatMap(numberFormatter.number(from:))?.doubleValue,
+            let path = delegate.pathAttributes["d"].flatMap(CGPath.parse(_:)),
+            let fillColor = delegate.pathAttributes["fill"]
+        else {
+            return nil
+        }
+
+        self.size = CGSize(width: width, height: height)
+        self.path = path
+        self.fillColor = UIColor(hexString: fillColor)
+        self.isEOFilled = delegate.pathAttributes["fill-rule"]?.lowercased() == "evenodd"
+    }
+}
+
+extension CGPath {
+    fileprivate static func parse(_ d: String) -> CGPath? {
+        let scanner = Scanner(string: d)
+        scanner.charactersToBeSkipped = CharacterSet(charactersIn: " ,\n\t\r")
+        scanner.locale = Locale(identifier: "en_US_POSIX")
+
+        let path = CGMutablePath()
+        var hasSubpath = false
+
+        func nextCommand() -> Character? {
+            while !scanner.isAtEnd {
+                let character = scanner.string[scanner.currentIndex]
+                if ("A"..."Z").contains(character) || ("a"..."z").contains(character) {
```

Comment:
I'd suggest https://developer.apple.com/documentation/foundation/characterset 

I think it's faster O(1) vs O(26)

Reply:
🤔 `CharacterSet` works on `Unicode.Scalar`, not `Character`. I think the inline `"A"..."Z"` resolves to `ClosedRange<Character>` (still `O(1)`) and not `ClosedRange<String>`. But I can make it unambiguous.